kingpin/actors/aws/test/integration_ecs.py

```diff
@@ -0,0 +1,195 @@
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_03a_start_service_no_count(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='03')
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_03b_scale_service(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='03',
+                                         count=1)
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_03c_scale_service_dont_wait(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='03',
+                                         count=5)
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_03d_delete_service(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='03',
+                                         delete=True)
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_04a_start_service_scaled(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='04',
+                                         count=1)
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_04b_scale_service_down(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='04',
+                                         count=0)
+        yield actor.execute()
+
+    @attr('aws', 'integration')
+    @testing.gen_test(timeout=60)
+    def integration_04c_delete_service(self):
+        actor = self._make_service_actor(*self._hanging_task(), suffix='04',
+                                         delete=True)
+        yield actor.execute()
+
+    def _make_task_actor(self, name, task, wait=True):
+        with self._tmp_file(task) as definition:
+            actor = ecs.RunTask(
+                'Run %s' % name,
+                {'region': self.region,
+                 'cluster': self.cluster,
+                 'task_definition': definition,
+                 'wait': wait})
+        return actor
+
+    def _make_service_actor(self, name, task, suffix='', delete=False,
+                            count=0):
+        state = 'present'
+        if delete:
+            state = 'absent'
+        with self._tmp_file(task) as definition:
+            actor = ecs.Service(
+                'Service %s with count %s' % (name, count),
+                {'region': self.region,
+                 'cluster': self.cluster,
+                 'task_definition': definition,
+                 'service_name': '%s_%s' % (name, suffix),
+                 'state': state,
+                 'count': count})
+        return actor
+
+    def _successful_task(self):
```

Comment:
I'm not sure of the value of this layout here... it just means I have to scroll up and down to read the code.